openssl-dynamic/src/main/c/sslcontext.c

```diff
@@ -1659,6 +1540,155 @@ TCN_IMPLEMENT_CALL(void, SSLContext, setCertVerifyCallback)(TCN_STDARGS, jlong c
     }
 }
 
+/**
+ * Returns an array containing all the X500 principal's bytes.
+ *
+ * Partly based on code from conscrypt:
+ * https://android.googlesource.com/platform/external/conscrypt/+/master/src/main/native/org_conscrypt_NativeCrypto.cpp
+ */
+static jobjectArray principalBytes(JNIEnv* e, const STACK_OF(X509_NAME)* names) {
+    jobjectArray array;
+    jbyteArray bArray;
+    int i;
+    int count;
+    int length;
+    unsigned char *buf;
+    X509_NAME* principal;
+
+    if (names == NULL) {
+        return NULL;
+    }
+
+    count = sk_X509_NAME_num(names);
+    if (count <= 0) {
+        return NULL;
+    }
+
+    array = (*e)->NewObjectArray(e, count, byteArrayClass, NULL);
+    if (array == NULL) {
+        return NULL;
+    }
+
+    for (i = 0; i < count; i++) {
+        principal = sk_X509_NAME_value(names, i);
+        buf = NULL;
+        length = i2d_X509_NAME(principal, &buf);
+        if (length < 0) {
+            if (buf != NULL) {
+                // We need to delete the local references so we not leak memory as this method is called via callback.
+                OPENSSL_free(buf);
```

Comment:
Yep... Will do some more cleanup rounds anyway

> Am 24.06.2016 um 19:33 schrieb Nathan Mittler <notifications@github.com>:
> 
> In openssl-dynamic/src/main/c/sslcontext.c:
> 
> > +        return NULL;
> > +    }
> > +
> > +    array = (*e)->NewObjectArray(e, count, byteArrayClass, NULL);
> > +    if (array == NULL) {
> > +        return NULL;
> > +    }
> > +
> > +    for (i = 0; i < count; i++) {
> > +        principal = sk_X509_NAME_value(names, i);
> > +        buf = NULL;
> > +        length = i2d_X509_NAME(principal, &buf);
> > +        if (length < 0) {
> > +            if (buf != NULL) {
> > +                // We need to delete the local references so we not leak memory as this method is called via callback.
> > +                OPENSSL_free(buf);
> if this won't have a very big impact, then maybe we just leave it as-is for now?
> 
> @normanmaurer WDYT?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>